invokeai/frontend/web/src/features/controlLayers/components/CanvasLayersFloatingToolbar.tsx

```diff
@@ -0,0 +1,78 @@
+import { ButtonGroup, Flex, IconButton, Tooltip } from '@invoke-ai/ui-library';
+import {
+  useAddControlLayer,
+  useAddInpaintMask,
+  useAddNewRegionalGuidanceWithARefImage,
+  useAddRasterLayer,
+  useAddRegionalGuidance,
+} from 'features/controlLayers/hooks/addLayerHooks';
+import { useCanvasIsBusy } from 'features/controlLayers/hooks/useCanvasIsBusy';
+import { useIsEntityTypeEnabled } from 'features/controlLayers/hooks/useIsEntityTypeEnabled';
+import { memo } from 'react';
+import { useTranslation } from 'react-i18next';
+import { PiEraserBold, PiPaintBrushBold, PiRectangleBold, PiShapesFill } from 'react-icons/pi';
+
+export const CanvasLayersFloatingToolbar = memo(() => {
+  const { t } = useTranslation();
+  const isBusy = useCanvasIsBusy();
+  const addInpaintMask = useAddInpaintMask();
+  const addRegionalGuidance = useAddRegionalGuidance();
+  const _addRegionalReferenceImage = useAddNewRegionalGuidanceWithARefImage();
+  const addRasterLayer = useAddRasterLayer();
+  const addControlLayer = useAddControlLayer();
+  const isRegionalGuidanceEnabled = useIsEntityTypeEnabled('regional_guidance');
+  const isControlLayerEnabled = useIsEntityTypeEnabled('control_layer');
+  const isInpaintLayerEnabled = useIsEntityTypeEnabled('inpaint_mask');
```

Comment:
I can't remember if there are any instances when raster layers might be disabled, but probably want to handle that case along with the other layer types.